lib/template.html

```diff
@@ -111,27 +93,56 @@
   {% endif %}
 {% endif %}
 
+<script type="application/ld+json">
+  {
+    "@context" : "http://schema.org",
+
+{% if page.seo and page.seo.type %}
+    "@type" : {{ page.seo.type | jsonify }},
+{% elsif page.url == "/" %}
+    "@type" : "WebSite",
+{% elsif page.id %}
+    "@type" : "BlogPosting",
+{% else %}
+    "@type" : "WebPage",
+{% endif %}
+
+{% if seo_site_title %}
+    "name" : {{ seo_site_title | jsonify }},
+{% endif %}
+
+{% if seo_page_title %}
+    "headline": {{ seo_page_title | jsonify }},
+{% endif %}
+
+{% if page.image %}
+    "image": {{ page.image | jsonify }},
+{% endif %}
+
+{% if page.date %}
+    "datePublished": {{ page.date | date_to_xmlschema | jsonify }},
+{% endif %}
+
+{% if page.description %}
+    "description": {{ page.description | jsonify }},
+{% endif %}
+
 {% if site.logo %}
-  <script type="application/ld+json">
-    {
-      "@context": "http://schema.org",
-      "@type": "Organization",
-      "url": {{ seo_url | jsonify }},
-      "logo": {{ site.logo | prepend: "/" | prepend: seo_url | jsonify }}
-    }
-  </script>
-{% endif %}
-
-{% if site.social %}
-  <script type="application/ld+json">
-    {
-      "@context" : "http://schema.org",
-      "@type" : "{% if site.social.type %}{{ site.social.type }}{% else %}person{% endif %}",
-      "name" : "{% if site.social.name %}{{ site.social.name }}{% else %}{{ seo_site_title }}{% endif %}",
-      "url" : {{ seo_url | jsonify }},
-      "sameAs" : {{ site.social.links | jsonify }}
-    }
-  </script>
+    "logo": {{ site.logo | prepend: "/" | prepend: seo_url | jsonify }},
 {% endif %}
 
+{% if page.seo and page.seo.links %}
+    "sameAs" : {{ page.seo.links | jsonify }},
+{% elsif page.url == "/" and site.social and site.social.links %}
```

Comment:
This feels like :tophat:Magic. I wasn't sure if `site.social.links` is an existing thing that we need to continue to support as a fallback.